src/components/PlayClassCard/PlayClassCard.js

```diff
@@ -1,26 +1,133 @@
 import React from "react";
-import "../PlayClassCard/PlayClassCard.css";
+import "./PlayClassCard.css";
 import { IoPlay, IoStop, IoPause, IoHeart } from "react-icons/io5";
-import { useState } from "react";
+import { useState, useRef } from "react";
+import { useTimer } from "react-use-precision-timer";
+import tibetanBowl from "../../assets/tibetanBowl.mp3";
+import bellChime from "../../assets/bellChime.mp3";
+import bigSingingBowl from "../../assets/bigSingingBowl.mp3";
+import oceanWaves from "../../assets/oceanWaves.mp3";
 
 export default function PlayClassCard({ classToPlay, toggleFavourite }) {
   const [isActive, setIsActive] = useState(0);
-  const { id, name, prepTime, duration, isFavourite } = classToPlay;
+  const {
+    id,
+    name,
+    prepTime,
+    duration,
+    intervalTime,
+    isFavourite,
+  } = classToPlay;
+  const [formattedTime, setFormattedTime] = useState("00:00:00");
+
+  const tibetanBowlRef = useRef();
+  const oceanRef = useRef();
+  const bellChimeRef = useRef();
+  const bigSingingBowlRef = useRef();
+
+  const prepTimer = useTimer({
+    delay: prepTime * 1000,
+    runOnce: true,
+    callback() {
+      mainTimer.start();
+      intervalTimer.start();
+      bellChimeRef.current.play();
+    },
+  });
+
+  const mainTimer = useTimer({
+    fireImmediately: true,
+    delay: 1000,
+
+    callback() {
+      if (mainTimer.isRunning()) {
+        setFormattedTime(getFormattedTime());
+
+        if (oceanRef.current.paused) {
+          oceanRef.current.play();
+        }
+      }
+
+      if (getTime().total >= duration - intervalTime) {
+        intervalTimer.stop();
+      }
+
+      if (getTime().total >= duration) {
+        mainTimer.stop();
+        oceanRef.current.pause();
+        bigSingingBowlRef.current.play();
+      }
+    },
+  });
+
+  const intervalTimer = useTimer({
+    fireImmediately: false,
+    delay: intervalTime * 1000,
+    callback() {
+      if (intervalTimer.isRunning()) {
+        tibetanBowlRef.current.play();
+      }
+    },
+  });
 
   function toggleHeartIcon() {
     toggleFavourite(id);
   }
 
   function onPauseClick() {
-    setIsActive(1);
+    if (prepTimer.isRunning()) {
+      prepTimer.pause();
+    } else {
+      setIsActive(1);
+      mainTimer.pause();
+      intervalTimer.pause();
+      oceanRef.current.pause();
+    }
   }
 
   function onPlayClick() {
     setIsActive(2);
```

Comment:
Some things here:
1. I don't think you need an extra state to check if is active, because you have all the information you need in the timers
2. But, in case you need it, what does `1` `2` and `3` mean? If you wan to use numbers for this, you could create a few constants above, like this: 
```js
const FIRST_STATE = 1; 
const SECOND_STATE = 2;
const THIRD_STATE = 3;
```
I just made the names ,of course you should name this with some sense, not just first,second or third.

Reply:
I did tried the timer but it was not showing exactly what I wanted so I presented the pr with this other solution. But you are right I will try again to use the timer or change my code to be more understandable. Tks!